Tests/PenguinParallelTests/NonBlockingThreadPoolTests.swift

```diff
@@ -163,9 +163,9 @@ public class ThreadCountingPlatform<Underlying: ConcurrencyPlatform>: Concurrenc
     runningThreadCount += 1
     return underlying.makeThread(name: name) { [unowned self, fn] () in
       fn()
-      lock.lock()
-      runningThreadCount -= 1
-      lock.unlock()
```

Comment:
Does this change belong in this PR?

Reply:
This change was required in order to compile on mac with the latest mac toolchains.

Reply:
Srsly, to *compile*?  This looks like a semantic change.

Reply:
Yup: https://github.com/saeta/penguin/runs/660788911